check_db_genesis_balances.py

```diff
@@ -0,0 +1,214 @@
+        balance_str = split_balance[0] + "000000000"
+    elif len(split_balance) == 2:
+        balance_str = split_balance[0] + split_balance[1].ljust(9, "0")
+    return int(balance_str)
+
+
+with open(args.config_file) as json_file:
+    ledger = json.load(json_file)
+
+ledger_accounts = ledger["ledger"]["accounts"]
+
+cur = conn.cursor()
+cur.execute(count_query)
+result = cur.fetchone()
+cur.close()
+count = result[0] if result else 0
+
+assert count == len(
+    ledger_accounts
+), f"Number of accounts in the JSON file ({len(ledger_accounts)}) does not match the number of accounts in the SQL query ({count})."
+
+
+def json_account_to_ledger_account(account) -> LedgerAccount:
+    return {
+        "pk": account["pk"],
+        "balance": normalize_balance(account["balance"]),
+        "timing": (
+            {
+                "initial_minimum_balance": normalize_balance(
+                    account["timing"]["initial_minimum_balance"]
+                ),
+                "cliff_time": int(account["timing"]["cliff_time"]),
+                "cliff_amount": normalize_balance(account["timing"]["cliff_amount"]),
+                "vesting_period": int(account["timing"]["vesting_period"]),
+                "vesting_increment": normalize_balance(
+                    account["timing"]["vesting_increment"]
+                ),
+            }
+            if "timing" in account
+            else None
+        ),
+        "delegate": account["delegate"] if "delegate" in account else None,
+        "nonce": account["nonce"] if "nonce" in account else None,
+        "receipt_chain_hash": (
+            account["receipt_chain_hash"] if "receipt_chain_hash" in account else None
+        ),
+    }
+
+
+all_accounts_match = True
+for acc in ledger_accounts:
+    account = json_account_to_ledger_account(acc)
+    cur = conn.cursor()
+    cur.execute(query, (account["pk"],))
+    result = cur.fetchone()
+    result = row_to_ledger_account(result)
+    cur.close()
+
+    messages = []
+    if account["pk"] != result["pk"]:
+        messages.append(f"pk: {account['pk']} != {result['pk']}")
+
+    if account["balance"] != result["balance"]:
+        messages.append(f"balance: {account['balance']} != {result['balance']}")
+
+    if account["timing"] != result["timing"]:
+        messages.append(f"timing:\n{account['timing']} !=\n{result['timing']}")
+
+    if account["delegate"] != result["delegate"]:
+        messages.append(f"delegate: {account['delegate']} != {result['delegate']}")
+        if not (
+            account["nonce"] == result["nonce"]
+            if account["nonce"]
+            else result["nonce"] == 0
+        ):
+            messages.append(f"nonce: {account['nonce']} != {result['nonce']}")
+
+    if len(messages) != 0:
+        all_accounts_match = False
+        print(f"Account with pk '{account['pk']}' does not match the SQL query result.")
+        for message in messages:
+            print(f"\n{message}")
+
+conn.close()
```

Comment:
It is always a risk closing connection like that as we can have some crash between. I don't have psycopg2 automaticallly closes connection unless you will access cursor like:
```
with psycopg2.connect(..) as conn:
    with conn.cursor() as curs:
       curs.execute(SQL)
```